src/pages/academy/gameSimulator/subcomponents/chapterSimulator/ChapterSimulatorTextLoader.tsx

```diff
@@ -1,25 +1,23 @@
 import 'ace-builds/webpack-resolver';
 
 import { Button, Tab, Tabs } from '@blueprintjs/core';
-import React from 'react';
+import { useState } from 'react';
 import { toTxtPath } from 'src/features/game/assets/TextAssets';
 import { toS3Path } from 'src/features/game/utils/GameUtils';
-
-type Props = {
-  storageName: string;
-  s3TxtFiles: string[];
-};
+import { StorageProps } from 'src/features/gameSimulator/GameSimulatorTypes';
+import {
+  createHeadersWithCors,
+  loadFileLocally
+} from 'src/features/gameSimulator/GameSimulatorUtils';
 
 /**
- * This component enables story writers to upload their txt file contents
- * to the browser, or load a file from S3 and store the txt contents
- * in the browser. So that GameManager can read from these txt files
+ * This component allows chapter text files to be loaded either from S3, or from the user's local device.
  *
- * @param storageName the field in browser storage where the loaded/fetched txt files get stored temporarily
- * @param s3TxtFiles the list of S3 txt files to choose from
+ * @param storageName The field within browser storage to temporarily store the loaded / fetched text file(s).
+ * @param s3TxtFiles List of all text assets on S3 to choose from.
  */
-function CheckpointTxtLoader({ storageName, s3TxtFiles }: Props) {
-  const [chosenFilename, setChosenFilename] = React.useState(s3TxtFiles[0]);
+function ChapterSimulatorTextLoader({ storageName, s3TxtFiles }: StorageProps) {
```

Comment:
Ditto on typing, also perhaps we can use lambda functions for code style consistency (nit).

Reply:
Addressed in [962b341](https://github.com/source-academy/frontend/pull/2836/commits/962b3413e413ff74052dd0e7f90ff5830ec709fb). Thanks for the catch! Forgot to convert to lambda 😅